### PR TITLE
Remove now-unused bitvector conversion exceptions

### LINKED ISSUE #2103: Extend boolean conversion to throw structured exceptions for other kinds of conversion error
`convert_bitvector` currently throws an exception when something goes wrong with flatten_byte_extraction operations. This should be extended for other kinds of errors.

### PR DESCRIPTION
**Pull request by tautschnig**
Byte-operator lowering no longer throws any exceptions, and thus don't
need to attempt to catch them either. Remove the entire infrastructure
built around these.

Fixes: #2103

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by smowton — APPROVED** [review]

File: src/util/throw_with_nested.h [hunk 1]
```diff
@@ -1,60 +0,0 @@
...
-
-#ifndef CPROVER_UTIL_THROW_WITH_NESTED_H
-#define CPROVER_UTIL_THROW_WITH_NESTED_H
-
-#include <exception>
-
-#ifdef _MSC_VER
-#include <stdexcept>
-// TODO(tkiley): Nested exception logging not supported on windows due to a bug
-// TODO(tkiley): in MSVC++ Compiler (diffblue/cbmc#2104):
```
Comment (thk123) [inline src/util/throw_with_nested.h:17]: Add `fixes #2104` to auto close the issue

File: src/solvers/flattening/boolbv.cpp [hunk 2]
```diff
@@ -171,9 +171,6 @@ bvt boolbvt::conversion_failed(const exprt &expr)
 /// \param expr: Expression to convert
 /// \return A vector of literals corresponding to the outputs of the Boolean
 ///   circuit
-/// \throws bitvector_conversion_exceptiont raised if converting byte_extraction
-/// goes wrong.
-/// TODO: extend for other types of conversion exception (diffblue/cbmc#2103).
```
Comment (thk123) [inline src/solvers/flattening/boolbv.cpp:176]: Add `fixes 2103` to commit

**Review by thk123 — APPROVED** [review]
I guess it was done in an earlier PR - this would have been easier to review I think if changes removing the throws were included too. But since this just removes apparently dead code lgtm.

File: src/util/throw_with_nested.h (hunk 1 above)
Comment (tautschnig) [reply src/util/throw_with_nested.h:17]: Done.

**Comment by tautschnig** [comment]
> I guess it was done in an earlier PR - this would have been easier to review I think if changes removing the throws were included too.

I have step-by-step removed the throws by actually implementing the cases, hence combining that into a single PR didn't seem like the best option. The more-or-less final step in getting rid of the throws was #4651.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 2b4f35ef14ef3664eb9ecdd1698955023a55a1e8).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113718309

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**